### PR TITLE
Convert commands(`agJ`, `anj`, `pi`) to rizin APIs

### PR DESCRIPTION
**Pull request by b1llow**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://cutter.re/docs/contributing/code/getting-started.html) to this repository
- [ ] I made sure to follow the project's [coding style](https://cutter.re/docs/contributing/code/development-guidelines.html)
- [ ] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**Detailed description**

- `agJ`
- `anj`
- `pi`

**Test plan (required)**

**Closing issues**

Partially addresses https://github.com/rizinorg/cutter/issues/2666

### CODE REVIEW

File: src/widgets/DisassemblerGraphView.cpp [hunk 1]
```diff
@@ -197,7 +189,19 @@ void DisassemblerGraphView::loadCurrentGraph()
         highlight_token = nullptr;
     }
 
-    emptyGraph = !functions.size();
+    RzAnalysisFunction *fcn = Core()->functionIn(seekable->getOffset());
+
+    windowTitle = tr("Graph");
+    QString fcn_name = fcn && RZ_STR_ISNOTEMPTY(fcn->name)
+            ? std::unique_ptr<char, decltype(std::free) *> { rz_str_escape_utf8_for_json(fcn->name,
+                                                                                         -1),
```
Comment (XVilka) [inline src/widgets/DisassemblerGraphView.cpp]: Formatting looks broken here

**Review by XVilka — CHANGES_REQUESTED** [review]
```
:\projects\cutter\build\include\librz\rz_util/rz_time.h(18): error C2332: 'struct': missing tag name
c:\projects\cutter\build\include\librz\rz_util/rz_time.h(18): error C2556: '*__timezone(void)': overloaded function differs only by return type from 'long *__timezone(void)'
C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt\time.h(74): note: see declaration of '__timezone'
c:\projects\cutter\build\include\librz\rz_util/rz_time.h(18): error C2371: '__timezone': redefinition; different basic types
C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt\time.h(74): note: see declaration of '__timezone'
c:\projects\cutter\build\include\librz\rz_util/rz_time.h(18): error C2491: '__timezone': definition of dllimport function not allowed
c:\projects\cutter\build\include\librz\rz_util/rz_time.h(24): error C2332: 'struct': missing tag name
c:\projects\cutter\build\include\librz\rz_util/rz_time.h(24): error C2143: syntax error: missing ',' before '*'
c:\projects\cutter\build\include\librz\rz_util/rz_time.h(24): error C2059: syntax error: '*'
ninja: build stopped: subcommand failed.
Command exited with code 1
```

**Comment by b1llow** [comment]
Maybe it's because of redefinition to `__timezone`, but it's `timezone` in `rz_time.h`, which is confusing
And I found that compiling rizin with msvc was successful,
But it fails to compile cutter's **cpp** file

**Review by XVilka — APPROVED** [review]

File: src/widgets/DisassemblerGraphView.cpp [hunk 2]
```diff
@@ -197,7 +189,20 @@ void DisassemblerGraphView::loadCurrentGraph()
         highlight_token = nullptr;
     }
 
-    emptyGraph = !functions.size();
+    RzAnalysisFunction *fcn = Core()->functionIn(seekable->getOffset());
+
+    windowTitle = tr("Graph");
+    QString fcn_name = "Empty";
```
Comment (XVilka) [inline src/widgets/DisassemblerGraphView.cpp]: The suggestion by @karliss applies here as well: https://github.com/rizinorg/cutter/pull/3012#discussion_r942455726

File: src/widgets/DisassemblerGraphView.cpp [hunk 3]
```diff
@@ -256,63 +252,79 @@ void DisassemblerGraphView::loadCurrentGraph()
             gb.edges.emplace_back(block_jump);
         }
 
-        CutterJson switchOp = block["switchop"];
-        if (switchOp.size()) {
-            for (CutterJson caseOp : switchOp["cases"]) {
-                RVA caseJump = caseOp["jump"].toRVA();
+        RzAnalysisSwitchOp *switch_op = bbi->switch_op;
```
Comment (XVilka) [inline src/widgets/DisassemblerGraphView.cpp]: Same here

File: src/widgets/DisassemblerGraphView.cpp [hunk 4]
```diff
@@ -256,63 +252,79 @@ void DisassemblerGraphView::loadCurrentGraph()
...
+        }
 
-            if (iterator != opArray.end()) {
+        RzCoreDisasmOptions options = {};
+        options.vec = vec.get();
+        options.cbytes = 1;
+        rz_core_print_disasm(core, bbi->addr, buf.get(), (int)bbi->size, (int)bbi->size, NULL,
+                             &options);
+
+        auto vec_visitor = CutterPVector<RzAnalysisDisasmText>(vec.get());
```
Comment (XVilka) [inline src/widgets/DisassemblerGraphView.cpp]: And here

File: src/core/Cutter.cpp [hunk 5]
```diff
@@ -777,34 +777,32 @@ void CutterCore::delFlag(const QString &name)
     emit flagsChanged();
 }
 
+PRzAnalysisBytes CutterCore::getRzAnalysisBytesSingle(RVA addr)
+{
+    CORE_LOCK();
+    ut8 buf[128];
+    rz_io_read_at(core->io, addr, buf, sizeof(buf));
+    std::unique_ptr<RzPVector, decltype(rz_pvector_free) *> vec {
```
Comment (XVilka) [inline src/core/Cutter.cpp:785]: @karliss, you are the C++ expert, is it possible to simplify this block?

**Review by XVilka — CHANGES_REQUESTED** [review]

**Review by XVilka — APPROVED** [review]

File: src/core/Cutter.cpp (hunk 5 above)
Comment (karliss) [reply src/core/Cutter.cpp:785]: I have a couple of ideas for some rizin C++ helpers which could make the cutter code interacting with it slightly cleaner, but that's more of a job for separate issue.

Comment (karliss) [reply src/core/Cutter.cpp:785]: Also my opinion that most Rizin functions which force caller to temporary change seek is bad API and indicator that those functions should probably have offset argument.

**Comment by karliss** [comment]
This PR will probably fix #2800 (although I am not sure about underlying cause of printing invalid json which would still be good to fix in rizin). Need to recheck after this is PR is merged.

File: src/core/Cutter.cpp (hunk 5 above)
Comment (XVilka) [reply src/core/Cutter.cpp:785]: Agree, we should open an issue about that in Rizin, probably worth fixing before 0.5.0

**Pull request merged**